src/main/java/ua/softserveinc/tc/service/impl/CalendarServiceImpl.java

```diff
@@ -133,15 +137,13 @@ public void add(Event event) {
                         roomDao.findById(recurrentEventDto.getRoomId()));
                 newRecurrentEvent.setColor(recurrentEventDto.getColor());
 
-                eventDao.create(newRecurrentEvent);
-                /// / TODO: 21.12.2016 throwsException??
-
-                res.add(genericMapper.toDto(newRecurrentEvent));
+                res.add(newRecurrentEvent);
             }
             calendar.add(Calendar.WEEK_OF_YEAR, 1);
             calendar.set(Calendar.DAY_OF_WEEK, daysOFWeek.get("Mon"));
         }
-        return res;
+        eventDao.saveSetOfEvents(res);
```

Comment:
It is good that you improve that as I and Vitaliy told yesterday. Good implementation our thoughts.